Source/Scene/Cesium3DTilesetTraversal.js

```diff
@@ -1,856 +1,673 @@
 define([
         '../Core/CullingVolume',
+        '../Core/defaultValue',
         '../Core/defined',
         '../Core/freezeObject',
         '../Core/Intersect',
         '../Core/ManagedArray',
         '../Core/Math',
         '../Core/OrthographicFrustum',
-        './Cesium3DTileChildrenVisibility',
+        './Cesium3DTileOptimizationHint',
         './Cesium3DTileRefine',
         './SceneMode'
     ], function(
         CullingVolume,
+        defaultValue,
         defined,
         freezeObject,
         Intersect,
         ManagedArray,
         CesiumMath,
         OrthographicFrustum,
-        Cesium3DTileChildrenVisibility,
+        Cesium3DTileOptimizationHint,
         Cesium3DTileRefine,
         SceneMode) {
     'use strict';
 
     /**
      * @private
      */
-    var Cesium3DTilesetTraversal = {};
+    function Cesium3DTilesetTraversal() {
+    }
+
+    var VisibilityFlag = {
+        NONE : 0,
+        VISIBLE : 1,
+        IN_REQUEST_VOLUME : 2
+    };
+
+    function isVisibleBit(flag) {
+        return (flag & VisibilityFlag.VISIBLE) > 0;
+    }
+
+    function inRequestVolumeBit(flag) {
+        return (flag & VisibilityFlag.IN_REQUEST_VOLUME) > 0;
+    }
 
-    function selectTiles(tileset, frameState, outOfCore) {
+    function clearVisibility(tile) {
+        tile._visibilityFlag = tile._visibilityFlag & ~VisibilityFlag.VISIBLE;
+    }
+
+    function isVisible(tile) {
+        var flag = tile._visibilityFlag;
+        return isVisibleBit(flag) && inRequestVolumeBit(flag);
+    }
+
+    function isVisibleButNotInRequestVolume(tile) {
+        var flag = tile._visibilityFlag;
+        return isVisibleBit(flag) && !inRequestVolumeBit(flag);
+    }
+
+    var traversal = {
```

Comment:
@lilleyse I'm curious, have you looked at the memory profile / GC pressure before/after this change when reading in the "steady state" - no new tiles being streamed?

Reply:
In both master and here I'm not seeing anything show up in Chrome's allocating sampling profile, and the memory usage in Chrome's task manager is staying steady.